### PR TITLE
Disable p2pd compilation by default

### PR DESCRIPTION
**Pull request by yhn112**
After these changes, both `pip install hivemind` and `pip install -e hivemind` will use precompiled binary p2pd, and will not require Go installation.
To run with building it from source, one should use `pip install hivemind --install-option="--buildgo"` (with `-e`, optionally).

Also, fixes a minor bug with the exception in case of no Go installed.

### CODE REVIEW

**Review by justheuristic — APPROVED** [review]
Good job!

**Review by deniskamazur — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -90,15 +86,39 @@ def libp2p_download_install():
...
 
 class Install(install):
+    user_options = install.user_options + [('buildgo', None, None)]
+
+    def initialize_options(self):
+        super().initialize_options()
+        self.buildgo = False
+
+    def finalize_options(self):
+        super().finalize_options()
```
Comment (mryab) [inline setup.py:96]: del

File: setup.py [hunk 2]
```diff
@@ -90,15 +86,39 @@ def libp2p_download_install():
 
 
 class Install(install):
+    user_options = install.user_options + [('buildgo', None, None)]
```
Comment (mryab) [inline setup.py:89]:
```suggestion
    user_options = install.user_options + [('build_go', None, None)]
```

**Review by mryab — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (yhn112) [reply setup.py]: Fixed

File: setup.py (hunk 2 above)
Comment (yhn112) [reply setup.py:89]: Impossible to apply

**Review by mryab — APPROVED** [review]

**Pull request merged**